src/player_activity.cpp

```diff
@@ -318,6 +318,11 @@ void player_activity::finish( player *p )
             p->do_read(&(p->i_at(position)));
             if (type == ACT_NULL) {
                 add_msg(_("You finish reading."));
+
+                // below code added in to support reading from ground/vehicles
+                if ( values[1] != FRM_INV ) {
+                    g->drop(p->inv.size()-1);
```

Comment:
Does `g->drop(position); ` work here?  I'm not 100% confident that the book will always be the last object in the inventory array, there may be edge-cases where another item is automatically added (if someone's smoking a cigarette when they start reading, I think it would add a cigarette butt to their inventory mid-action).

Also this feature is fantastic, thanks for doing it!